sbt-scala-native/src/main/scala/scala/scalanative/sbtplugin/testinterface/ComRunner.scala

```diff
@@ -24,27 +24,29 @@ class ComRunner(bin: File,
                 args: Seq[String],
                 logger: Logger) {
 
-  private[this] val serverSocket = new ServerSocket(0)
-  private[this] val port         = serverSocket.getLocalPort
-
   private[this] val runner = new Thread {
     override def run(): Unit = {
       import sbt.Process._
-      running = true
+      val port = serverSocket.getLocalPort
       logger.info(s"Starting process '$bin' on port '$port'.")
       Process(bin.toString +: port.toString +: args, None, envVars.toSeq: _*) ! logger
-      running = false
-      socket.close()
-      serverSocket.close()
     }
   }
 
-  private[this] var running: Boolean = false
+  private[this] var serverSocket: ServerSocket = _
+  private[this] val socket =
+    try {
+      serverSocket = new ServerSocket(0)
 
-  runner.start()
+      runner.start()
 
-  serverSocket.setSoTimeout(30 * 1000)
-  private[this] val socket = serverSocket.accept()
+      serverSocket.setSoTimeout(30 * 1000)
+      serverSocket.accept()
+    } catch {
+      case _: SocketTimeoutException =>
+        throw new MessageOnlyException(
+          "The test program never connected to sbt.")
+    } finally serverSocket.close()
```

Comment:
Sounds reasonable. Lets document it with a comment that explains why we close it so soon.